src/app/services/widget-service/widget.service.ts

```diff
@@ -69,14 +82,31 @@ export class WidgetService {
     return slideConfig;
   }
 
-  disabledButtons(showWidget: number, weatherWidgets: WeatherWidget[], btnLeft: ElementRef, btnRight: ElementRef, removeLast: MatButton) {
+  removeLast(showWidget: number, weatherWidgets: WeatherWidget[], btnLeft: ElementRef, btnRight: ElementRef, removeLast: MatButton, snackBar : MatSnackBar) {
```

Comment:
Too many parameters for functions. You need to split one big function into a couple small ones.